test/internal/packagemanifests.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Modified from https://github.com/kubernetes-sigs/kubebuilder/tree/39224f0/test/e2e/v3
+
+package internal
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+)
+
+var makefilePackagemanifestsFragment = `
+# Options for "packagemanifests".
+ifneq ($(origin FROM_VERSION), undefined)
+PKG_FROM_VERSION := --from-version=$(FROM_VERSION)
+endif
+ifneq ($(origin CHANNEL), undefined)
+PKG_CHANNELS := --channel=$(CHANNEL)
+endif
+ifeq ($(IS_CHANNEL_DEFAULT), 1)
+PKG_IS_DEFAULT_CHANNEL := --default-channel
+endif
+PKG_MAN_OPTS ?= $(FROM_VERSION) $(PKG_CHANNELS) $(PKG_IS_DEFAULT_CHANNEL)
+
+# Generate package manifests.
+packagemanifests: kustomize manifests
+	operator-sdk generate kustomize manifests -q --interactive=false
+	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
+	$(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests -q --version $(VERSION) $(PKG_MAN_OPTS)
+`
+
+// AddPackagemanifestsTarget will append the packagemanifests target to the makefile
+// in order to test the steps described in the docs.
+// More info:  https://v1-0-x.sdk.operatorframework.io/docs/olm-integration/generation/#package-manifests-formats
+func (tc TestContext) AddPackagemanifestsTarget() error {
+	makefileBytes, err := ioutil.ReadFile(filepath.Join(tc.Dir, "Makefile"))
+	if err != nil {
+		return err
+	}
+
+	b, err := ioutil.ReadFile(filepath.Join(tc.Dir, "PROJECT"))
+	if err != nil {
+		return err
+	}
+	c := &config.Config{}
+	if err = c.Unmarshal(b); err != nil {
+		return err
+	}
+
+	if !strings.HasPrefix(c.Layout, "go") {
+		// The target is equals for all types but NonGo projects has not the manifests target which
+		// needs to be replaced here.
+		makefilePackagemanifestsFragment = strings.Replace(makefilePackagemanifestsFragment,
+			"packagemanifests: kustomize manifests", "packagemanifests: kustomize", 1)
```

Comment:
This is brittle. If we change the packagemanifests target to something slightly different, this replacement will silently fail.

Instead, what if we make the fragment a format string that we pass to `fmt.Sprintf()`. e.g.

```go
makefilePackagemanifestsFragment := `
# Options for "packagemanifests".
ifneq ($(origin FROM_VERSION), undefined)
PKG_FROM_VERSION := --from-version=$(FROM_VERSION)
endif
ifneq ($(origin CHANNEL), undefined)
PKG_CHANNELS := --channel=$(CHANNEL)
endif
ifeq ($(IS_CHANNEL_DEFAULT), 1)
PKG_IS_DEFAULT_CHANNEL := --default-channel
endif
PKG_MAN_OPTS ?= $(FROM_VERSION) $(PKG_CHANNELS) $(PKG_IS_DEFAULT_CHANNEL)
# Generate package manifests.
packagemanifests: %s
	operator-sdk generate kustomize manifests -q --interactive=false
	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
	$(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests -q --version $(VERSION) $(PKG_MAN_OPTS)
`

pmDeps := []string{"kustomize"}
if strings.HasPrefix(c.Layout, "go") {
	pmDeps = append(pmDeps, "manifests")
}
fragment := fmt.Sprintf(makefilePackagemanifestsFragment, strings.Join(pmDeps, " "))
```

Reply:
It is good always check. You are right here.